src/helpers/load-script.ts

```diff
@@ -0,0 +1,72 @@
+/**
+ * Holds a list of scripts that are still loading.
+ */
+let loadingQueue: string[] = [];
```

Comment:
suggestion: please rename to `scriptLoadingQueue` and therefore remove the comment.